src/SFA.DAS.Reservations.Web/Controllers/ReservationsController.cs

```diff
@@ -61,9 +61,19 @@ public async Task<IActionResult> PostApprenticeshipTraining(ReservationsRouteMod
             StartDateModel startDateModel = null;
             if (!string.IsNullOrWhiteSpace(formModel.TrainingStartDate))
                 startDateModel = JsonConvert.DeserializeObject<StartDateModel>(formModel.TrainingStartDate);
+
             Course course = null;
-            if (!string.IsNullOrWhiteSpace(formModel.SelectedCourse))
-                course = JsonConvert.DeserializeObject<Course>(formModel.SelectedCourse);
+
```

Comment:
again here just assign the course id and let the handler do the heavy lifting, no need to look up the course here.

Reply:
Moved the logic to a new handler